uaclient/util.py

```diff
@@ -806,3 +807,34 @@ def parse_rfc3339_date(dt_str: str) -> datetime.datetime:
     return datetime.datetime.strptime(
         dt_str_without_microseconds, "%Y-%m-%dT%H:%M:%S%z"
     )
+
+
+def validate_proxy(
+    protocol: str, proxy: Optional[str], test_url: str
+) -> Optional[str]:
+    if not proxy:
+        return None
+
+    if not is_service_url(proxy):
+        print("{} is not a valid url. Not setting as proxy.".format(proxy))
```

Comment:
When encountering invalid I'd prefer the approach of stopping the line with loud UserFacingErrorr/log errors. In cloud-init we are too flexible and try as hard as we can to ignore non-critical errors, which leads to much later recognition that something is misconfigured on the system. The more strict we are, the more time is saves a customer having invalid config recognized.

Reply:
Yep this is something I wanted to bring up/talk about - this was just my first draft. I do like the "fail fast and loud" approach when appropriate.

Thinking about it more, I think it makes sense here to fail loudly, because just like you say, if we continue after this error then we will try to use entitlements without the proxies in working order. The user specifically configured ua to use a proxy, so continuing operations while effectively ignoring the proxy would be unexpected/undesirable. 

We can chat more about this but I'll probably go ahead and change it to raise a UserFacingError